xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -607,8 +607,31 @@ void CLinuxRendererGLES::RenderUpdateVideo(bool clear, DWORD flags, DWORD alpha)
     CDVDVideoCodecIMXBuffer *buffer = m_buffers[m_iYV12RenderBuffer].IMXBuffer;
     if (buffer != NULL && buffer->IsValid())
     {
-      ManageDisplay();
-      g_IMXContext.SetBlitRects(m_sourceRect, m_destRect);
+      RENDER_STEREO_MODE stereo_mode = g_graphicsContext.GetStereoMode();
+      if (stereo_mode)
+        g_graphicsContext.SetStereoView(RENDER_STEREO_VIEW_LEFT);
+
+     ManageDisplay();
+
+     if (stereo_mode = g_graphicsContext.GetStereoMode())
+       g_graphicsContext.SetStereoView(RENDER_STEREO_VIEW_OFF);
+
+     CRect srcRect(m_sourceRect);
+     CRect dstRect(m_destRect);
+     switch (stereo_mode)
+     {
+       case RENDER_STEREO_MODE_SPLIT_HORIZONTAL:
+         // fix me
```

Comment:
Whenever I call ManageDisplay, the Gui layer's view port is moved to 0,960,1920,1080 or something. This I can cope with when doing L|R with just multiplying the .x2 * 2 (code in RENDER_STEREO_MODE_SPLIT_VERTICAL).

When I call ManageDisplay and then try to do Top / Bottom - I cannot get for the hell of it, the left eye over the other. It just stretches L.

Whenever I ommit ManageDisplay and only apply x2*2.0 respectively y2*2.0 the movie works as expected. But the Subs viewport is not set correctly.

The current code does L|R correctly including subs.